src/main/java/baritone/process/ChestSortProcess.java

```diff
@@ -0,0 +1,629 @@
+    }
+
+    @Override
+    public void activate() {
+        this.active = true;
+    }
+
+    @Override
+    public PathingCommand onTick(boolean calcFailed, boolean isSafeToCancel) {
+        if (scanner == null && sorter == null) { // uninitialized
+            this.scanWorld(this.ctx.world());
+        }
+
+        if (isChestOpen(ctx)) {
+            this.getVisitor().containerOpenTick((ContainerChest)ctx.player().openContainer);
+            if (!this.getVisitor().wantsContainerOpen()) {
+                ctx.player().closeScreenAndDropStack();
+            }
+        }
+
+        while (this.getVisitor().finished()) {
+            if (this.getVisitor() == this.scanner) {
+                this.setSorting(SortingChestVisitor.fromScanner(this, this.scanner));
+            } else {
+                this.onLostControl(); // sorter has finished
+                return null;
+            }
+        }
+
+        final ChestVisitor visitor = getVisitor();
+        TileEntityChest target = visitor.getCurrentTarget().get();
+        Goal goal = new GoalNear(target.getPos(), 2);
+        Optional<Rotation> newRotation = getRotationForChest(target);
+        newRotation.ifPresent(rotation -> {
+            baritone.getLookBehavior().updateTarget(rotation, true);
+        });
+        final RayTraceResult trace = ctx.objectMouseOver();
+        if (!(isChestOpen(ctx)) && getConnectedChests(target).stream().anyMatch(chest -> ctx.isLookingAt(chest.getPos()))) {
+            ctx.playerController().processRightClickBlock(ctx.player(), ctx.world(), target.getPos(), trace.sideHit, trace.hitVec, EnumHand.OFF_HAND);
+        }
+
+        return new PathingCommand(goal, PathingCommandType.REVALIDATE_GOAL_AND_PATH);
+    }
+
+    @Override
+    public void onLostControl() {
+        this.active = false;
+
+        // reset the state
+        // if onLostControl is called we assume that the state of the chests have changed and we should start from scratch
+        this.phase = null;
+        this.scanner = null;
+        this.sorter = null;
+    }
+
+    @Override
+    public String displayName0() {
+        return "ChestSortProcess";
+    }
+
+    private void setScanning(ScanningChestVisitor scanner) {
+        this.phase = Phase.SCANNING;
+        this.scanner = scanner;
+        this.sorter = null;
+    }
+    private void setSorting(SortingChestVisitor sorter) {
+        this.phase = Phase.SORTING;
+        this.sorter = sorter;
+        this.scanner = null;
+    }
+
+    private ChestVisitor getVisitor() {
+        switch (this.phase) {
+            case SCANNING: return Objects.requireNonNull(this.scanner);
+            case SORTING:  return Objects.requireNonNull(this.sorter);
+
+            default: throw new IllegalStateException();
+        }
+    }
+
+    private void scanWorld(World world) {
+        List<TileEntityChest> allChests = world.loadedTileEntityList
+            .stream()
+            .filter(ent -> ent instanceof TileEntityChest)
```

Comment:
imagine not doing `.filter(TileEntityChest.class::isInstance).map(TileEntityChest.class::cast)`